src/main/java/com/project/springbootproject/model/ShoppingCart.java

```diff
@@ -0,0 +1,48 @@
+package com.project.springbootproject.model;
+
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.JoinColumn;
+import jakarta.persistence.OneToMany;
+import jakarta.persistence.OneToOne;
+import jakarta.persistence.Table;
+import java.util.Set;
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import org.apache.commons.lang3.builder.ToStringExclude;
+import org.hibernate.annotations.SQLDelete;
+import org.hibernate.annotations.SQLRestriction;
+
+@Data
+
+@Entity
+
+@Table(name = "shopping_carts")
+
+@SQLDelete(sql = "UPDATE shopping_carts SET is_deleted = TRUE WHERE id = ?")
+
+@SQLRestriction("is_deleted = FALSE")
+
+public class ShoppingCart {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @OneToOne
+    @JoinColumn(name = "user_id")
+    @ToStringExclude
+    @EqualsAndHashCode.Exclude
+    private User user;
```

Comment:
I'd recommend to use user's id for cart instead of generated value as they are tightly connected.
Please check `@MapsId` annotation and usage of it